Automattic/woocommerce-payments
# Fraud Risk Tools: Add an account data based feature flag

Fixes https://github.com/Automattic/woocommerce-payments-server/issues/3124

#### Changes proposed in this Pull Request
This PR: 
- updates the feature flag name to match what we've done for other flags.
- Adds a way for she feature flag to be defined on the server (account data) and the local site options.







#### Testing instructions
##### Testing the options fallback.
- Reset the settings: `npm run wp option delete  _wcpay_fraud_protection_settings_enabled wcpay_fraud_protection_settings_active`
- Open the [overview page](http://localhost:8082/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Foverview) (localhost link) and inspect the browser console. See that the value for `wcpaySettings.isFraudProtectionSettingsEnabled` is empty.
- Turn on the fallback by running: `npm run wp option update _wcpay_fraud_protection_settings_enabled 1
`
- Open the [overview page](http://localhost:8082/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Foverview) (localhost link) and inspect the browser console. See that the value for `wcpaySettings.isFraudProtectionSettingsEnabled` is `1`.

##### Testing the server account data
- Clear the options again: `npm run wp option delete  _wcpay_fraud_protection_settings_enabled`
-  Open the [overview page](http://localhost:8082/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Foverview) (localhost link) and inspect the browser console. See that the value for `wcpaySettings.isFraudProtectionSettingsEnabled` is empty.
- Apply this patch to the server:
```php
diff --git a/server/wp-content/rest-api-plugins/endpoints/wcpay/class-accounts-controller.php b/server/wp-content/rest-api-plugins/endpoints/wcpay/class-accounts-controller.php
index 37ec9acc8..ab0eb0780 100644
--- a/server/wp-content/rest-api-plugins/endpoints/wcpay/class-accounts-controller.php
+++ b/server/wp-content/rest-api-plugins/endpoints/wcpay/class-accounts-controller.php
@@ -703,12 +703,13 @@ class Accounts_Controller extends Base_Controller {
 				'primary_color'   => $stripe_account->get_branding_primary_color(),
 				'secondary_color' => $stripe_account->get_branding_secondary_color(),
 			],
-			'locale'                           => $account->locale,
-			'has_more_failed_events'           => $this->webhook_service->does_account_have_pending_events( $account, $account->is_test ),
-			'has_submitted_vat_data'           => false !== $this->vat_service->get_vat_data( $account ),
-			'is_documents_enabled'             => VAT_Utils::is_supported_country( $account->country ),
-			'card_testing_protection_eligible' => $this->account_service->calculate_card_testing_prevention_eligibility( $account ),
-			'progressive_onboarding'           => $this->onboarding_service->get_po_info( $account ),
+			'locale'                            => $account->locale,
+			'has_more_failed_events'            => $this->webhook_service->does_account_have_pending_events( $account, $account->is_test ),
+			'has_submitted_vat_data'            => false !== $this->vat_service->get_vat_data( $account ),
+			'is_documents_enabled'              => VAT_Utils::is_supported_country( $account->country ),
+			'card_testing_protection_eligible'  => $this->account_service->calculate_card_testing_prevention_eligibility( $account ),
+			'progressive_onboarding'            => $this->onboarding_service->get_po_info( $account ),
+			'is_fraud_protection_settings_enabled' => '1',
 		];
 
 		if ( ! $client_supports_deposits_blocked_status ) {

```
- Clear your account cache. Reload the dev tools to make sure it is fetched again.
- Scroll to the bottom to see if the value is in the account data:
<img width="614" alt="Screenshot 2023-03-20 at 18 54 34" src="https://user-images.githubusercontent.com/1713474/226412338-2f54e73a-7d02-47c4-a53b-8cadfae9cdff.png">

- Open the [overview page](http://localhost:8082/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Foverview) (localhost link) and inspect the browser console. See that the value for `wcpaySettings.isFraudProtectionSettingsEnabled` is `1`.

-------------------


**Post merge**
